doc/devel/coding_guide.rst

```diff
@@ -1,369 +1,318 @@
-.. _pr-guidelines:
```

Comment:
Is this content copied 1:1 tp the new `pr_guide.rst`? If so, it would be much easier to understand, if you make a `git mv` of the file. Note that this needs its own commit. If you move a file and create the old file with new content in one commit, git claims you modified the old file and created a new.